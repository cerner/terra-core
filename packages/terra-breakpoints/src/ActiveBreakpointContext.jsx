import React from 'react';

const ActiveBreakpointContext = React.createContext();

export default ActiveBreakpointContext;
