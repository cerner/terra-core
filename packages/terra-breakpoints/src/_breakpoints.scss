@import './variables';

:export {
  @each $property, $value in $terra-breakpoints {
    #{$property}: $value;
  }
}
