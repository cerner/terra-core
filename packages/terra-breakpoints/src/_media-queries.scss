@import './variables';

@mixin terra-mq-small-up {
  @media screen and (min-width: #{map-get($terra-breakpoints, small)}px) {
    @content;
  }
}

@mixin terra-mq-medium-up {
  @media screen and (min-width: #{map-get($terra-breakpoints, medium)}px) {
    @content;
  }
}

@mixin terra-mq-large-up {
  @media screen and (min-width: #{map-get($terra-breakpoints, large)}px) {
    @content;
  }
}

@mixin terra-mq-huge-up {
  @media screen and (min-width: #{map-get($terra-breakpoints, huge)}px) {
    @content;
  }
}

@mixin terra-mq-enormous-up {
  @media screen and (min-width: #{map-get($terra-breakpoints, enormous)}px) {
    @content;
  }
}
