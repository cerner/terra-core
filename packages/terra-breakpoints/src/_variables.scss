$terra-breakpoints: (
  tiny: 0,
  small: 544,
  medium: 768,
  large: 992,
  huge: 1216,
  enormous: 1440
);
