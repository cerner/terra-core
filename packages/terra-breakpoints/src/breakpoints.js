const breakpoints = {
  tiny: 0,
  small: 544,
  medium: 768,
  large: 992,
  huge: 1216,
  enormous: 1440,
};

const breakpointSequence = ['tiny', 'small', 'medium', 'large', 'huge', 'enormous'];

const activeBreakpointForSize = (size) => {
  for (let i = breakpointSequence.length - 1; i >= 0; i -= 1) {
    const breakpoint = breakpointSequence[i];

    if (size >= breakpoints[breakpoint]) {
      return breakpoint;
    }
  }
  return undefined;
};

const breakpointIsActiveForSize = (breakpoint, size) => {
  const breakpointLocation = breakpointSequence.indexOf(breakpoint);
  const sizeLocation = breakpointSequence.indexOf(activeBreakpointForSize(size));

  return sizeLocation === -1 || breakpointLocation === -1 ? false : sizeLocation >= breakpointLocation;
};

export default breakpoints;
export {
  activeBreakpointForSize, breakpointIsActiveForSize,
};
