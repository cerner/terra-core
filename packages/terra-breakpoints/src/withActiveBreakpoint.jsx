import React from 'react';
import ActiveBreakpointContext from './ActiveBreakpointContext';

const withActiveBreakpoint = (WrappedComponent) => {
  const WithActiveBreakpointComp = props => (
    <ActiveBreakpointContext.Consumer>
      {activeBreakpoint => <WrappedComponent {...props} activeBreakpoint={activeBreakpoint} />}
    </ActiveBreakpointContext.Consumer>
  );

  WithActiveBreakpointComp.displayName = `withActiveBreakpoint(${WrappedComponent.displayName || WrappedComponent.name || 'Component'})`;
  WithActiveBreakpointComp.WrappedComponent = WrappedComponent;

  return WithActiveBreakpointComp;
};

export default withActiveBreakpoint;
