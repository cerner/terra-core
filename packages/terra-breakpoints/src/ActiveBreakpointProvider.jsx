import React from 'react';
import PropTypes from 'prop-types';
import ActiveBreakpointContext from './ActiveBreakpointContext';
import { activeBreakpointForSize } from './breakpoints';

const propTypes = {
  /**
   * The component(s) that will be wrapped by `<ActiveBreakpointContext.Provider />`.
   */
  children: PropTypes.node,
};

class ActiveBreakpointProvider extends React.Component {
  constructor(props) {
    super(props);

    this.setActiveBreakpoint = this.setActiveBreakpoint.bind(this);

    this.state = {
      activeBreakpoint: activeBreakpointForSize(window.innerWidth),
    };
  }

  componentDidMount() {
    window.addEventListener('resize', this.setActiveBreakpoint);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.setActiveBreakpoint);
  }

  setActiveBreakpoint() {
    const { activeBreakpoint } = this.state;
    const newBreakpoint = activeBreakpointForSize(window.innerWidth);

    if (activeBreakpoint !== newBreakpoint) {
      this.setState({
        activeBreakpoint: newBreakpoint,
      });
    }
  }

  render() {
    const { children } = this.props;
    const { activeBreakpoint } = this.state;

    return (
      <ActiveBreakpointContext.Provider value={activeBreakpoint}>
        {children}
      </ActiveBreakpointContext.Provider>
    );
  }
}

ActiveBreakpointProvider.propTypes = propTypes;

export default ActiveBreakpointProvider;
