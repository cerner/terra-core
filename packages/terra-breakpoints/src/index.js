import breakpoints, { activeBreakpointForSize, breakpointIsActiveForSize } from './breakpoints';
import ActiveBreakpointContext from './ActiveBreakpointContext';
import ActiveBreakpointProvider from './ActiveBreakpointProvider';
import withActiveBreakpoint from './withActiveBreakpoint';

export default breakpoints;
export {
  activeBreakpointForSize,
  breakpointIsActiveForSize,
  ActiveBreakpointContext,
  ActiveBreakpointProvider,
  withActiveBreakpoint,
};
