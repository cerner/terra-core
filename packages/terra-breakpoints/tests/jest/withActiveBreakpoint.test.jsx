import React from 'react';
import { withActiveBreakpoint } from '../../src/index';

describe('withActiveBreakpoint', () => {
  // Snapshot Tests
  it('should render using the provided render function', () => {
    const TestComponent = withActiveBreakpoint(() => <div>I am a test component</div>);

    const wrapper = mount(<TestComponent />);
    expect(wrapper).toMatchSnapshot();
  });
});
