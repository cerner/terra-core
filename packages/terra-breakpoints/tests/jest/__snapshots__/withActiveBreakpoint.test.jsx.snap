// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withActiveBreakpoint should render using the provided render function 1`] = `
<withActiveBreakpoint(Component)>
  <Component>
    <div>
      I am a test component
    </div>
  </Component>
</withActiveBreakpoint(Component)>
`;
